frontend/src/app/shared/components/autocompleter/project-menu-autocomplete/project-menu-autocomplete.component.ts

```diff
@@ -27,305 +27,124 @@
 //++
 
 import { PathHelperService } from 'core-app/core/path-helper/path-helper.service';
-import {
-  IAutocompleteItem,
-  ILazyAutocompleterBridge,
-} from 'core-app/shared/components/autocompleter/lazyloaded/lazyloaded-autocompleter';
-import { KeyCodes } from 'core-app/shared/helpers/keyCodes.enum';
-import { isClickedWithModifier } from 'core-app/shared/helpers/link-handling/link-handling';
 import { I18nService } from 'core-app/core/i18n/i18n.service';
-import { HttpClient } from '@angular/common/http';
 import {
-  ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, OnInit,
+  ChangeDetectionStrategy,
+  Component,
 } from '@angular/core';
 import { CurrentProjectService } from 'core-app/core/current-project/current-project.service';
-
-export interface IProjectMenuEntry {
-  id:number;
-  name:string;
-  identifier:string;
-  parents:IProjectMenuEntry[];
-  level:number;
-}
-
-export type ProjectAutocompleteItem = IAutocompleteItem<IProjectMenuEntry>;
+import { combineLatest } from 'rxjs';
+import {
+  debounceTime,
+  map,
+  shareReplay,
+} from 'rxjs/operators';
+import { IProject } from 'core-app/core/state/projects/project.model';
+import { insertInList } from 'core-app/shared/components/project-include/insert-in-list';
+import { IProjectData } from 'core-app/shared/components/project-list/project-data';
+import { recursiveSort } from 'core-app/shared/components/project-include/recursive-sort';
```

Comment:
All three of the above also have some overlap with the code I wrote for the project autocompleters, though they are a bit different. It might be useful to take a look at consolidating them now.

e.g. `frontend/src/app/shared/components/autocompleter/project-autocompleter/insert-in-list.ts` or `frontend/src/app/shared/components/autocompleter/project-autocompleter/recursive-sort.ts` are very similar but not exactly.

Reply:
As discussed, we will tackle that separately.